### PR TITLE
adding support to search ami from ami catalog and copying ami

### PR DESCRIPTION
**Pull request by ashwini3326**
@rohanpm @JAVGan PTAL. This mere request introduce two changes - 

1. Search Image in AMI Catalog which includes images from community too.
2. Copy AMI from community to account of self.

### CODE REVIEW

File: cloudimg/aws.py [hunk 1]
```diff
@@ -240,6 +240,32 @@ def get_image_by_filters(self, filters):
...
         return self.ec2.Image(images[0]['ImageId'])
 
+    def get_image_from_ami_catalog(self, image_id):
+        """
+        Finds Image in AMI Catalog with given filters.
+        The search includes AMIs owned by self, AWS marketplace AMIs,
+        and Community AMIs.
+
+        Args:
+            filters(list): List of filters (dict) to apply.
```
Comment (rohanpm) [inline cloudimg/aws.py:250]: This doesn't match the method's actual signature, there is no list of filters (dict).

File: cloudimg/aws.py [hunk 2]
```diff
@@ -632,6 +658,25 @@ def import_snapshot(self, obj, snapshot_name, tags=None):
 
         return snapshot
 
+    def copy_ami(self, image_id, image_name, image_region, tags=None):
```
Comment (rohanpm) [inline cloudimg/aws.py:661]: What's the `tags` argument about?  It's not documented and not used.

File: cloudimg/aws.py [hunk 3]
```diff
@@ -632,6 +658,25 @@ def import_snapshot(self, obj, snapshot_name, tags=None):
...
+        """
+        Create copy of an AMI.
+
+        Args:
+            image_id (str): AMI Id of the image to copy
+            name (str): Name of new image
+            region(str): Region for new Image.
+
+        Returns:
+            Dict with Image_id of the newly created AMI and requests_id
```
Comment (rohanpm) [inline cloudimg/aws.py:671]: Is this true?  The variable `obj_name` sounds like it would be a `str` containing a name, not a `dict` containing an ID.

**Review by rohanpm — CHANGES_REQUESTED** [review]

File: cloudimg/aws.py (hunk 2 above)
Comment (ashwini3326) [reply cloudimg/aws.py:661]: Did it for testing, missed to remove it. Deleting now.

File: cloudimg/aws.py (hunk 3 above)
Comment (ashwini3326) [reply cloudimg/aws.py:671]: It returns a Dict with Image Id and Response Metadata. I am changing the name of return value. 
Screenshot from 2024-04-04 14-36-31

https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CopyImage.html#API_CopyImage_ResponseElements

File: cloudimg/aws.py (hunk 1 above)
Comment (ashwini3326) [reply cloudimg/aws.py]: fixed it.

**Review by rohanpm — APPROVED** [review]

**Review by JAVGan — APPROVED** [review]

**Comment by JAVGan** [comment]
LGTM, able to be merged whenever possible

**Pull request merged**